zerver/lib/user_status.py

```diff
@@ -125,10 +154,65 @@ def get_user_status(user_profile: UserProfile) -> UserInfoDict:
             "emoji_name",
             "emoji_code",
             "reaction_type",
+            "status_end_time",
         )
         .first()
     )
 
     if not status_set_by_user:
         return {}
     return format_user_status(status_set_by_user)
+
+
+def maybe_update_scheduled_status(
+    user_profile: UserProfile,
+    client_id: int,
+) -> None:
+    try:
+        scheduled_user_status = ScheduledUserStatus.objects.get(
+            user_profile=user_profile,
+            realm=user_profile.realm,
+            client_id=client_id,
+        )
+        scheduled_user_status.delete()
+    except ScheduledUserStatus.DoesNotExist:
+        return None
+
+
+def clear_user_status(user_status: ScheduledUserStatus) -> None:
+    update_user_status(
+        user_profile=user_status.user_profile,
+        status_text="",
+        client_id=user_status.client_id,
+        emoji_name="",
+        emoji_code="",
+        reaction_type=UserStatus.UNICODE_EMOJI,
+        status_end_time="",
+    )
+    event = dict(
+        type="user_status",
+        user_id=user_status.user_profile.id,
+        status_text="",
+        emoji_name="",
+        emoji_code="",
+        reaction_type=UserStatus.UNICODE_EMOJI,
+    )
+    send_event(
+        user_status.user_profile.realm,
+        event,
+        get_user_ids_who_can_access_user(user_status.user_profile),
+    )
+
+
+@transaction.atomic
+def try_clear_scheduled_user_status() -> bool:
+    user_status_to_clear = (
+        ScheduledUserStatus.objects.filter(scheduled_timestamp__lte=timezone_now())
```

Comment:
Can't we just use `UserStatus` model to manage everything. I think we can do this with much simplicity using only the `UserStatus` model. We already store the expiry time there.